compiler/parser/RdrHsSyn.hs

```diff
@@ -2253,6 +2255,16 @@ mkAppTy ty1 ty2 = noLoc $ HsAppTy noExt ty1 ty2
 unitType :: LHsType GhcPs
 unitType = noLoc $ HsTupleTy noExt HsBoxedOrConstraintTuple []
 
+-- Maybe types and values
```

Comment:
These are for the object language, right? Then they should be `Optional`, `Some` and `None`.

Reply:
yup, tha's fixed